include/votca/xtp/IncrementalFockBuilder.h

```diff
@@ -0,0 +1,106 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+#ifndef VOTCA_XTP_INCREMENTALFOCKBUILDER_H
+#define VOTCA_XTP_INCREMENTALFOCKBUILDER_H
+
+#include "votca/xtp/logger.h"
+#include <votca/tools/types.h>
+namespace votca {
+namespace xtp {
+
+// Small Wrapper class to build incremental fock matrix
+class IncrementalFockBuilder {
+ public:
+  IncrementalFockBuilder(Logger& log, double start_threshold,
+                         Index fock_matrix_reset)
+      : log_(log),
+        start_incremental_F_threshold_(start_threshold),
+        fock_matrix_reset_(fock_matrix_reset) {}
+
+  void Configure(const Eigen::MatrixXd& dmat) {
+    Ddiff_ = dmat;
+    Dlast_ = dmat;
+  }
+
+  void Start(Index iteration, double DiisError) {
+    if (!incremental_Fbuild_started_ &&
+        DiisError < start_incremental_F_threshold_) {
+      incremental_Fbuild_started_ = true;
+      reset_incremental_fock_formation_ = false;
+      last_reset_iteration_ = iteration - 1;
+      next_reset_threshold_ = DiisError / 10.0;
+      XTP_LOG(Log::error, log_)
+          << TimeStamp() << " Using incremental 4c build from here"
+          << std::flush;
+    }
```

Comment:
I now see how you use it in the loop, but I don't understand the logic now, it seems that as soon as you have called `Start` `incremental_Fbuild_started` will be `true` and will not be turned to false again? In that case this function block only executes once, so why not call it once?

Reply:
Because we want to start it in the middle of the iterations, when the threshold is reached and not before we start the SCF. So each SCF cycle before it starts it has to be checked if it should start.